File: src/test/resources/corpus/canonico/quick-sort.java
class Solucao {
    void ordena(int[] v, int inicio, int fim) {
        if (inicio >= fim) return;
        int pivo = particiona(v, inicio, fim);
        ordena(v, inicio, pivo - 1);
        ordena(v, pivo + 1, fim);
    }

    private int particiona(int[] v, int inicio, int fim) {
        int referencia = v[fim];
        int i = inicio - 1;
        for (int j = inicio; j < fim; j++) {
            if (v[j] <= referencia) {
                i++;
                int tmp = v[i];
                v[i] = v[j];
                v[j] = tmp;
            }
        }
        int tmp = v[i + 1];
        v[i + 1] = v[fim];
        v[fim] = tmp;
        return i + 1;
    }
}

File: src/test/resources/corpus/c/canonico/quick-sort.c
int particionar(int valores[], int inicio, int fim) {
    int pivo = valores[fim];
    int menor = inicio - 1;

    for (int j = inicio; j < fim; j++) {
        if (valores[j] <= pivo) {
            menor++;
            int guarda = valores[menor];
            valores[menor] = valores[j];
            valores[j] = guarda;
        }
    }

    int guarda = valores[menor + 1];
    valores[menor + 1] = valores[fim];
    valores[fim] = guarda;
    return menor + 1;
}

void ordenarRapido(int valores[], int inicio, int fim) {
    if (inicio >= fim) {
        return;
    }
    int corte = particionar(valores, inicio, fim);
    ordenarRapido(valores, inicio, corte - 1);
    ordenarRapido(valores, corte + 1, fim);
}
